cabal-install/Distribution/Client/ProjectConfig/Legacy.hs

```diff
@@ -1098,6 +1119,22 @@ legacyPackageConfigFieldDescrs =
       , "fail-when-no-test-suites", "test-wrapper" ]
   . commandOptionsToFields
   ) (testOptions' ParseArgs)
+ ++
+  ( liftFields
+      legacyBenchmarkFlags
+      (\flags conf -> conf { legacyBenchmarkFlags = flags })
+  . mapFieldNames
+      prefixBenchmark
+  . addFields
+      [ newLineListField "benchmark-options"
+          (showTokenQ . fromPathTemplate) (fmap toPathTemplate parseTokenQ)
+          benchmarkOptions
+          (\v conf -> conf { benchmarkOptions = v })
+      ]
+  . filterFields
+      []
```

Comment:
This can't be right. That's just going to give you `[]`. So either just ignore `benchmarkOptions'` by removing the whole `commandOptionsToFields` stuff below and do the `addFields` like you do now or remove the `addFields` and let `"benchmark-options"` through the filter.

Reply:
Ok, I think I get what the code is doing now. So the addField is necessary because of the special quoting rules for config files as opposed to on the command line. That's what the `parseTokenQ`/`showTokenQ` things do.

So yeah, using filterFields is not right here.

In principle you definetly can get rid of the  `commandOptionsToFields` conversion. Remember that this whole thing is just constructing a list of field descriptors, so instead of the filterFields, addFields, etc. you could just construct the list right off the bat. In the interest of keeping this extensible I suppose it is better not to do that though.